### DISCUSSION TITLE
What is the design purpose of job name?

### DISCUSSION THREAD

**Discussion by vilicvane** [taskforcesh/bullmq#1448] [Q&A] [answered]
Is it designed as a descriptive field for debugging purpose or a matching field that affects program behavior?

I am feeling the former way (considering the later one can be achieved through job data), but [documentation](https://docs.bullmq.io/patterns/named-processor) suggests the later.

**Comment by manast** [comment, marked as answer]
You can use it for debugging, or as the documentation exemplifies, for controlling behavior.
